src/dsp_tools/commands/xmllib/models/root.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from dataclasses import field
+from pathlib import Path
+
+from lxml import etree
+
+from dsp_tools.commands.excel2xml import append_permissions
+from dsp_tools.commands.excel2xml import write_xml
+from dsp_tools.commands.xmllib.models.resource import Resource
+
+XML_NAMESPACE_MAP = {None: "https://dasch.swiss/schema", "xsi": "http://www.w3.org/2001/XMLSchema-instance"}
+DASCH_SCHEMA = "{https://dasch.swiss/schema}"
+
+
+@dataclass
+class XMLRoot:
+    shortcode: str
+    default_ontology: str
+    resource_collection: list[Resource] = field(default_factory=list)
+
+    def make_root(self) -> etree._Element:
+        schema_url = (
+            "https://raw.githubusercontent.com/dasch-swiss/dsp-tools/main/src/dsp_tools/resources/schema/data.xsd"
+        )
+        schema_location_key = str(etree.QName("http://www.w3.org/2001/XMLSchema-instance", "schemaLocation"))
+        schema_location_value = f"https://dasch.swiss/schema {schema_url}"
+        return etree.Element(
+            f"{DASCH_SCHEMA}knora",
+            attrib={
+                schema_location_key: schema_location_value,
+                "shortcode": self.shortcode,
+                "default-ontology": self.default_ontology,
+            },
+            nsmap=XML_NAMESPACE_MAP,
+        )
+
+    def add_resources(self, resources: list[Resource]) -> XMLRoot:
+        self.resource_collection.extend(resources)
+        return self
```

Comment:
for convenience and consistency sake, this could also expose `add_resource` (singular) and `add_resource_optional`, which would add only a couple of lines, and shouldn't be confusing to the users, if the same three variants are exposed on all values

Reply:
I agree mostly but I don't see how optional can be used?

Reply:
if you have a function that looks at some data and may or may not return a resource. I am aware that you don't think it's likely that anyone from RDU would write such code, but as I said, it doesn't hurt to add it, so I think a small likelihood and API consistency are good enough reasons.

Reply:
what would the input look like though?

Reply:
the input to what?